manager/orchestrator/updater.go

```diff
@@ -111,9 +117,17 @@ func (u *Updater) Run(ctx context.Context, cluster *api.Cluster, service *api.Se
 	}
 	// Abort immediately if all tasks are clean.
 	if len(dirtyTasks) == 0 {
+		if service.UpdateStatus != nil && service.UpdateStatus.State == api.UpdateStatus_RUNNING {
+			u.completeUpdate(ctx, service.ID)
+		}
 		return
 	}
 
+	// If there's no update in progress, we are starting one.
+	if service.UpdateStatus == nil || service.UpdateStatus.State == api.UpdateStatus_NEW {
+		u.startUpdate(ctx, service.ID)
```

Comment:
So, the update supervisor keeps a map of `Updater` around and cancels the previous one when it detects an update.

Maybe the `Updater` should expose the `TaskSpec` it's updating to and the `UpdateSupervisor` should not cancel an update if it's already working towards the same goal?